connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -507,15 +497,54 @@ private WorkerTask buildWorkerTask(ClusterConfigState configState,
             TransformationChain<SinkRecord> transformationChain = new TransformationChain<>(connConfig.<SinkRecord>transformations(), retryWithToleranceOperator);
             SinkConnectorConfig sinkConfig = new SinkConnectorConfig(plugins, connConfig.originalsStrings());
             retryWithToleranceOperator.reporters(sinkTaskReporters(id, sinkConfig, errorHandlingMetrics));
+
+            Map<String, Object> props = consumerConfig(connConfig, id);
+            KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props);
+
             return new WorkerSinkTask(id, (SinkTask) task, statusListener, initialState, config, configState, metrics, keyConverter,
                     valueConverter, headerConverter, transformationChain, loader, time,
-                    retryWithToleranceOperator);
+                    retryWithToleranceOperator, consumer);
         } else {
             log.error("Tasks must be a subclass of either SourceTask or SinkTask", task);
             throw new ConnectException("Tasks must be a subclass of either SourceTask or SinkTask");
         }
     }
 
+    private Map<String, Object> producerConfigs() {
```

Comment:
Maybe add the `ConnectorConfig` and `ConnectorTaskId` like in `consumerConfig(...)` to be more consistent?